src/routes/conversation/[id]/upload-pdf/+server.ts

```diff
@@ -0,0 +1,41 @@
+import { authCondition } from "$lib/server/auth";
+import { collections } from "$lib/server/database";
+import { MAX_SEQ_LEN as CHUNK_CAR_LEN, createEmbeddings } from "$lib/server/embeddings";
+import { uploadPdfEmbeddings } from "$lib/server/files/uploadFile";
+import { chunk } from "$lib/utils/chunk";
+import { error } from "@sveltejs/kit";
+import { ObjectId } from "mongodb";
+import * as pdfjsLib from "pdfjs-dist/legacy/build/pdf";
+
+export async function POST({ request, params, locals }) {
+	const conversationId = new ObjectId(params.id);
+	const conversation = await collections.conversations.findOne({
+		_id: conversationId,
+		...authCondition(locals),
+	});
+
+	if (!conversation) {
+		throw error(404, "Conversation not found");
+	}
+
+	const formData = await request.formData();
+	const file = formData.get("pdf"); // 'pdf' is the name used in FormData on the frontend
+	const data = new Uint8Array(await file.arrayBuffer());
```

Comment:
```suggestion
	if (!file || typeof file === "string") {
		throw error(400, "No file provided");
	}
	const data = new Uint8Array(await file.arrayBuffer());
```

Check that pdf is indeed a file (for type checks)